### PR TITLE
[FLINK-8086][kafka] Ignore ProducerFencedException

### PR DESCRIPTION
**Pull request by pnowojski**
ProducerFencedException can happen if we restore twice from the same checkpoint
    or if we restore from an old savepoint. In both cases transactional.ids that we
    want to recoverAndCommit have been already committed and reused. Reusing mean
    that they will be known by Kafka's brokers under newer producerId/epochId,
    which will result in ProducerFencedException if we try to commit again some
    old (and already committed) transaction.
    
    Ignoring this exception might hide some bugs/issues, because instead of failing
    we might have a semi silent (with a warning) data loss.

## Verifying this change

This change added a `testFailAndRecoverSameCheckpointTwice` for a scenario that was previously not tested (and was failing).

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java [hunk 1]
```diff
@@ -720,11 +719,10 @@ protected void abort(KafkaTransactionState transaction) {
...
 		switch (semantic) {
 			case EXACTLY_ONCE:
-				FlinkKafkaProducer<byte[], byte[]> producer =
-					initTransactionalProducer(transaction.transactionalId, false);
-				producer.resumeTransaction(transaction.producerId, transaction.epoch);
-				producer.abortTransaction();
-				producer.close();
+				try (FlinkKafkaProducer<byte[], byte[]> producer =
+						initTransactionalProducer(transaction.transactionalId, false)) {
+					producer.initTransactions();
```
Comment (aljoscha) [inline flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java]: why is this changed from
```
resumeTransaction()
abortTransaction()
```
to
```
initTransaction()
```

Comment (pnowojski) [reply flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java]: `initTransaction()` does the same thing as resuming and aborting previous transactions, but in a safer way, since `resumeTransaction()` is part of our `KafkaProducer`s extension.

Comment (aljoscha) [reply flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java]: I see, thanks! I will merge this now. 😃

**Comment by pnowojski** [comment]
Thanks :)
